ipc/provider/src/checkpoint.rs

```diff
@@ -89,20 +89,59 @@ impl<T: BottomUpCheckpointRelayer + Send + Sync + 'static> BottomUpCheckpointMan
     }
 
     /// Run the bottom up checkpoint submission daemon in the background
-    pub fn run(self, validator: Address, submission_interval: Duration) {
-        tokio::spawn(async move {
-            loop {
-                if let Err(e) = self.submit_checkpoint(&validator).await {
-                    log::error!("cannot submit checkpoint for validator: {validator} due to {e}");
-                }
-
-                tokio::time::sleep(submission_interval).await;
+    pub async fn run(self, validator: Address, submission_interval: Duration) {
+        loop {
+            if let Err(e) = self.submit_checkpoint(&validator).await {
+                log::error!("cannot submit checkpoint for validator: {validator} due to {e}");
             }
-        });
+
+            tokio::time::sleep(submission_interval).await;
+        }
     }
 
     /// Submit the checkpoint from the target validator address
     pub async fn submit_checkpoint(&self, validator: &Address) -> Result<()> {
```

Comment:
We should remove `validator` from all the functions as the argument (I think I missed it in my previous review). We don't need to be a `validator` to act as a relayer in the system.
```suggestion
    pub async fn submit_checkpoint(&self, relayer: &Address) -> Result<()> {
```